app/src/main/java/com/pr0gramm/app/sync/SyncService.kt

```diff
@@ -32,10 +27,10 @@ class SyncService(private val userService: UserService,
     init {
         AsyncScope.launch {
             userService.loginStates
-                    .mapNotNull { state -> state.uniqueToken }
+                    .filter { state -> state.authorized }
```

Comment:
Das funktioniert so nicht. Hat aber vorher auch nicht ordentlich funktioniert. Wenn der User sich a ab- und wieder anmeldet, wird nur "true" durch gelassen. Mit dem distinctUntilChanged gibt's nach der ersten Anmeldung keine weiteren emits. Wäre vielleicht besser, wenn man ein mapNotNull auf den Usernamen macht, dann distinct und dann ein Filter hinterher.

Reply:
Ich hätte jetzt das hier gemacht:
```suggestion
                    .mapNotNull { state -> state.name }
```

Also dass der Rest dann so aussieht:
```kt
            userService.loginStates
                    .mapNotNull { state -> state.name }
                    .distinctUntilChanged()
                    .onStart { delay(1.seconds) }
                    .collect { launch { performSyncSeenService() } }
````
Nach dem `distinctUntilChanged` bleibt ja nur ein `name` über, wonach sollte man den filtern?

Reply:
Ne beides nicht optimal. Geht darum, dass der Sync automatisch getriggered wird, wenn sich der User State ändert. Ja passiert vermutlich auch so, aber naja...
Über die `id` gehts, haste recht:

```kotlin
            userService.loginStates
                    .map { state -> state.id }
                    .distinctUntilChanged()
                    .onStart { delay(1.seconds) }
                    .collect { token -> launch { performSyncSeenService() } }```

Reply:
Ich hab das in 72f0f6c mal gemacht. Dabei habe ich auch aus:
```kt
.collect { token -> launch { performSyncSeenService() } }
```
das `token` entfernt:
```kt
.collect { launch { performSyncSeenService() } }
```
Passt das so?